packages/noya-state/src/reducers/application.ts

```diff
@@ -459,263 +517,38 @@ export function reducer(
-          const style = layer.style;
-
-          if (!style) return;
-
-          switch (action[0]) {
-            case 'deleteDisabledBorders':
-              if (style.borders) {
-                style.borders = style.borders.filter(
-                  (border) => border.isEnabled,
-                );
-              }
-              break;
-            case 'deleteDisabledFills':
-              if (style.fills) {
-                style.fills = style.fills.filter((fill) => fill.isEnabled);
-              }
-              break;
-            case 'deleteDisabledShadows':
-              if (style.shadows) {
-                style.shadows = style.shadows.filter((fill) => fill.isEnabled);
-              }
-              break;
-          }
-        });
-      });
-    }
+    case 'deleteDisabledShadows':
     case 'setBorderColor':
     case 'setFillColor':
-    case 'setShadowColor': {
-      const [, index, color] = action;
-      const pageIndex = getCurrentPageIndex(state);
-      const layerIndexPaths = getSelectedLayerIndexPaths(state);
-
-      return produce(state, (state) => {
-        accessPageLayers(state, pageIndex, layerIndexPaths).forEach((layer) => {
-          const style = layer.style;
-
-          if (!style) return;
-
-          switch (action[0]) {
-            case 'setBorderColor':
-              if (style.borders && style.borders[index]) {
-                style.borders[index].color = color;
-              }
-              break;
-            case 'setFillColor':
-              if (style.fills && style.fills[index]) {
-                style.fills[index].color = color;
-              }
-              break;
-            case 'setShadowColor':
-              if (style.shadows && style.shadows[index]) {
-                style.shadows[index].color = color;
-              }
-              break;
-          }
-        });
-      });
-    }
-    case 'setBorderWidth': {
-      const [, index, amount, mode = 'replace'] = action;
-      const pageIndex = getCurrentPageIndex(state);
-      const layerIndexPaths = getSelectedLayerIndexPaths(state);
-
-      return produce(state, (state) => {
-        accessPageLayers(state, pageIndex, layerIndexPaths).forEach((layer) => {
-          const style = layer.style;
-
-          if (style && style.borders && style.borders[index]) {
-            const newValue =
-              mode === 'replace'
-                ? amount
-                : style.borders[index].thickness + amount;
-
-            style.borders[index].thickness = Math.max(0, newValue);
-          }
-        });
-      });
-    }
-    case 'setFillOpacity': {
-      const [, index, amount, mode = 'replace'] = action;
+    case 'setShadowColor':
+    case 'setBorderWidth':
+    case 'setFillOpacity':
+    case 'setOpacity':
+    case 'setShadowX':
+    case 'setShadowY':
+    case 'setShadowBlur':
+    case 'setBorderPosition':
+    case 'setShadowSpread': {
```

Comment:
These cases now delegate to the `styleReducer` function. We should do the same with the version of these for shared layer styles